cmake/Dependencies.cmake

```diff
@@ -170,3 +170,11 @@ endif()
 if(BUILD_docs)
   find_package(Doxygen)
 endif()
+
+# ---[ NCCL
+if(USE_NCCL)
+  add_definitions(-DUSE_NCCL)
+  find_package(NCCL REQUIRED)
```

Comment:
Now I can't build Caffe with CUDA unless I have NCCL. I'd like to see this behavior (same as with USE_CUDNN):

| USE_NCCL  | found NCCL? | use?
| ------------- | ------------- | ------------- |
| `On` | Yes | Yes
| `On` | No | ERROR
| *unset* | Yes | Yes
| *unset* | No | No
| `Off` | * | No